lib/ansible/plugins/callback/foreman.py

```diff
@@ -196,26 +207,24 @@ def send_reports(self, stats):
             metrics["time"] = {"total": int(time.time()) - self.start_time}
             now = datetime.now().strftime(self.TIME_FORMAT)
             report = {
-                "report": {
+                "config_report": {
                     "host": host,
                     "reported_at": now,
                     "metrics": metrics,
                     "status": status,
                     "logs": log,
                 }
             }
-            # To be changed to /api/v2/config_reports in 1.11.  Maybe we
-            # could make a GET request to get the Foreman version & do
-            # this automatically.
             try:
-                r = requests.post(url=self.FOREMAN_URL + '/api/v2/reports',
+                r = requests.post(url=self.FOREMAN_URL + '/api/v2/config_reports',
                                   data=json.dumps(report),
                                   headers=self.FOREMAN_HEADERS,
                                   cert=self.FOREMAN_SSL_CERT,
                                   verify=self.ssl_verify)
                 r.raise_for_status()
             except requests.exceptions.RequestException as err:
-                print(to_text(err))
+                self._display.warning('Sending report to Foreman at {url} failed for {host}: {err}'.format(
+                    host=host, err=to_text(err), url=self.FOREMAN_URL))
```

Comment:
We should probably use native strings here:
```suggestion
                    host=host, err=to_native(err), url=self.FOREMAN_URL))
```

cc @abadger

Reply:
No.  On the controller (callback plugins run on the controller), always use text strings.

But that does mean you should mark the format string with a u"".  There's some information about when that is important here: https://anonbadger.wordpress.com/2016/01/05/python2-string-format-and-unicode/

Reply:
@abadger I checked the source of that `warning()` function and it seems like it concatenates input with native strings which concerns me.

Reply:
yeah, please fix warning and deprecated.  Those need to use text strings when they concatenate.  (They really should use text strings everywhere and if I was fixing it, I would change all the literal strings into text strings but the concatenation is only thing I saw that causes an actual problem due to the various ways that Python2 and Python3 handle string formatting).

The way warning and deprecated are now, they could either traceback or cause tracebacks inside of display() when used with non-ascii strings of the wrong time.